ext/standard/basic_functions.c

```diff
@@ -2624,3 +2624,19 @@ PHP_FUNCTION(sys_getloadavg)
 }
 /* }}} */
 #endif
+
+/* {{{ */
+PHP_FUNCTION(nproc)
+{
+	ZEND_PARSE_PARAMETERS_NONE();
+
+	#if defined(_SC_NPROCESSORS_ONLN)
```

Comment:
This is Linux only so it should be extended to support other platforms as well if possible. For example on Mac this might work https://stackoverflow.com/a/22330309 and there will be probably something similar on FreeBSD. Windows should be also checked if there is a way to get it. If you can't test those platform, it can be checked in pipelines and other devs can possibly check.

Reply:
Good point. 

Took me a while to figure out how to configure windows to compile php. 
Updated the patch now to read the same information for windows. 

On FreeBSD and MacOS `_SC_NPROCESSORS_ONLN` is also available :tada: 
Is there another os to check?

Reply:
We still support Solaris and some other BSD variants like OpenBSD so maybe check those but it won't be huge issue if those are not initially supported by this.

Reply:
And here is a similar code for OpenBSD: https://github.com/openbsd/src/blob/8370bf70c786bf5f8c3e53fb889a14ffef8cff1d/lib/libc/gen/sysconf.c#L467-L470

Reply:
ok I think it's good in that case.